source/features/mark-private-orgs.js

```diff
@@ -1,38 +1,19 @@
-import {h} from 'dom-chef';
 import select from 'select-dom';
 import {getUsername} from '../libs/utils';
 import * as icons from '../libs/icons';
 import * as api from '../libs/api';
 
 export default async function () {
-	let publicOrgs = await api.v3(`users/${getUsername()}/orgs`);
-	if (!publicOrgs) {
-		return;
-	}
-	publicOrgs = publicOrgs.map(orgData => `/${orgData.login}`);
-
-	const userContainer = select('[itemtype="http://schema.org/Person"]');
-	if (!userContainer) {
-		return;
-	}
+	const publicOrgs = await api.v3(`users/${getUsername()}/orgs`);
 
-	// Find all org avatars
-	const orgAvatars = select.all('[itemprop="follows"]', userContainer);
-	for (const orgAvatar of orgAvatars) {
-		// Check if org is private
-		const orgPath = orgAvatar.getAttribute('href');
-		if (!orgPath) {
-			continue;
-		}
+	// Select organization avatars except public ones
+	const orgSelector = '.avatar-group-item[data-hovercard-type="organization"]';
+	const privateOrgSelector = orgSelector + publicOrgs
+		.map(orgData => `:not([href="/${orgData.login}"])`)
```

Comment:
@bfred-it How fast/slow will this method be compared to repetitively querying the DOM? Or just querying everything at once and filtering them in JS like before?

Because the selector might get pretty long for people who are part of many public organizations.

Reply:
Try it out:

```js
var not = Array(100).fill().map(a => `:not([href="${Math.random()}"])`).join('');
console.time('Selector');
select.all('.avatar' + not);
console.timeEnd('Selector');
```

The API is paginated to 100 so that's the most organizations you'd get.

On my MacBook 2013 (2.3 GHz Intel Core i7) that code runs in 0.5ms

Reply:
Yeah, I get that it should not matter because how modern browsers are efficient enough to do that kind of tasks.

But I thought that querying everything and filtering would be understood easily.

Maybe that's just me.

Reply:
Yeah it turns out `Array.includes` is shorter and clearer than the selector construction.